### PR TITLE
chore(launcher): revert meadata to custom properties & artifact name -> event path

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Reasons:
* in the future, when MLMD supports filtering, only top level fields in custom properties will be supported, so we'd better keep using them
* move artifact name to event path field, because 

```
// Event path is conceptually artifact name for the execution.
// We cannot store the name as a property of artifact "a", because for example:
// 1. In first task "preprocess", there's an output artifact "processed_data".
// 2. In second task "train", there's an input artifact "dataset" passed from "preprocess"
// task's "processed_data" output.
//
// Now the same artifact is called "processed_data" in "preprocess" task, but "dataset" in
// "train" task, because artifact name is related to the context it's used.
// Therefore, we should store artifact name as a property of the artifact's events
// (connects artifact and execution) instead of the artifact's property.
```
**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by zijianjoy** [comment]
/lgtm

Thank you Yuan for the change!

**Comment by zijianjoy** [comment]
/test kubeflow-pipelines-samples-v2

The previous test's sub-pipeline seems finished successfully https://4e18c21c9d33d20f-dot-datalab-vm-staging.googleusercontent.com/#/runs/details/cbbb47bf-f409-4943-a2a9-387f55e942b8, but parent-pipeline didn't get the run status.

File: samples/test/two_step_test.py [hunk 1]
```diff
@@ -32,66 +33,62 @@ def get_tasks(mlmd_connection_config, argo_workflow_name: str):
...
+            'artifacts': [],
+            'parameters': {
+                'some_int': 12,
+                'uri': 'uri-to-import'
+            }
+        },
+        'outputs': {
+            'artifacts': [{
+                'metadata': {},
+                'name': 'output_dataset_one',
```
Comment (zijianjoy) [inline samples/test/two_step_test.py:61]: dataset_one instead of output_dataset_one

**Comment by Bobgy** [comment]
Thanks @zijianjoy! I fixed it

**Comment by Bobgy** [comment]
/approve
Getting the big changes in, feel free to keep reviewing this.

**Pull request merged**